voluptuous/schema_builder.py

```diff
@@ -430,7 +430,9 @@ def validate_mapping(path, iterable, out):
 
                     break
                 else:
-                    if remove_key:
+                    if error:
+                        errors.append(error)
+                    elif remove_key:
```

Comment:
This looks like it introduced a regression and vol.Remove doesn't work anymore

I think it needs to be reversed
```
                    if remove_key:
                        # remove key
                        continue
                    elif error:
                        errors.append(error)
```